src/core/Akka/Event/DeadLetterListener.cs

```diff
@@ -65,112 +59,80 @@ private void IncrementCount()
                 _count++;
             }
         }
-
-        /// <summary>
-        /// TBD
-        /// </summary>
-        /// <param name="message">TBD</param>
-        /// <returns>TBD</returns>
         protected override bool Receive(object message)
         {
             if (_isAlwaysLoggingDeadLetters)
             {
-                return ReceiveWithAlwaysLogging()(message);
+                return ReceiveWithAlwaysLogging(message);
             }
 
             return Context.System.Settings.LogDeadLettersSuspendDuration != Timeout.InfiniteTimeSpan
                 ? ReceiveWithSuspendLogging(Context.System.Settings.LogDeadLettersSuspendDuration)(message)
-                : ReceiveWithMaxCountLogging()(message);
+                : ReceiveWithMaxCountLogging(message);
         }
 
-        private Receive ReceiveWithAlwaysLogging()
+        private bool ReceiveWithAlwaysLogging(object message)
         {
-            return message =>
-            {
-                if (message is AllDeadLetters d)
-                {
-                    if (!IsWrappedSuppressed(d))
-                    {
-                        IncrementCount();
-                        LogDeadLetter(d, "");
-                    }
-                    return true;
-                }
-                return false;
-            };
+            if (message is not AllDeadLetters d) return false;
+            if (IsWrappedSuppressed(d)) return true;
+            IncrementCount();
+            LogDeadLetter(d, "");
+            return true;
         }
 
-        private Receive ReceiveWithMaxCountLogging()
+        private bool ReceiveWithMaxCountLogging(object message)
```

Comment:
refactoring to be just a method, instead of allocating a new `Receive` each time